lib/dao.js

```diff
@@ -1526,6 +1526,27 @@ function NumberType(val) {
   return !isNaN(num) ? num : val;
 }
 
+function coerceArray(val) {
+  if (Array.isArray(val)) {
+    return val;
+  }
+
+  if (!utils.isPlainObject(val)) {
+    throw new Error(g.f('Value is not an {{array}} or {{array-like object}}'));
+  }
+
+  var arrayVal = new Array(Object.keys(val).length);
```

Comment:
Fortunately, V8 seems smart enough to allocate a sparse array where empty items do not take any (or very little) memory.

However, notice that the maximum array size is limited:

```
> new Array(5000000000)
RangeError: Invalid array length
```

In general, I am not comfortable overriding `qs` decisions about security in LoopBack. I mean if the team working on `qs` believe that automatically converting indexed query parameters into an array is a security vulnerability, then I trust them and don't want to undo their security-hardening measurements in loopback. If you believe they are wrong and it's ok to allow arbitrarily large arrays, then please talk to them and convince them to remove this restriction.